src/charm.py

```diff
@@ -126,6 +146,8 @@ def _render_mount_unit(self, path, target_path, extra_opts=None):
             systemd.service_stop(NFS_MOUNT_UNIT_NAME)
 
         mount_opts = "soft,async,fsc"
+        if mount_type == "efs":
+            mount_opts += ",tls"
```

Comment:
If we use EFS, do we always require TLS?

Reply:
No I have tested we can run without tls as well and the code has been updated to reflect the same.